### PR TITLE
fix(tools): add better description for vectorsearch tool and a test

### PR DESCRIPTION
**Pull request by douglas-reid**
OpenAI functions for QASearch weren't being reliably called. This creates a better tool description (the human description is used in the functions-based agent). A test is added to validate continued ability to work.

### CODE REVIEW

File: src/steamship/agents/tools/question_answering/vector_search_learner_tool.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ class VectorSearchLearnerTool(VectorSearchTool):
     name: str = "VectorSearchLearnerTool"
     human_description: str = "Learns a new fact and puts it in the Vector Database."
     agent_description: str = (
-        "Used to remember a fact. Only use this tool if someone asks to remember or learn something. ",
+        "Used to remember a fact. Only use this tool if someone asks to remember or learn something.",
```
Comment (eob) [inline src/steamship/agents/tools/question_answering/vector_search_learner_tool.py:18]: I think I wrote this original description with a bug.

In Python:

("S1", "S2", "S3") is a tuple of strings, but
("S1" "S2" "S3") is a concatenation of strings.

So:

* The commas in this list probably need to get removed
* The space after something. should be added back

Then the result will be the single string that was intended.

Alternative would be we use a different method of string concat that's less error prone -- my bad!

File: src/steamship/agents/tools/question_answering/vector_search_qa_tool.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ class VectorSearchQATool(VectorSearchTool):
     """Tool to answer questions with the assistance of a vector search plugin."""
 
     name: str = "VectorSearchQATool"
-    human_description: str = "Answers questions with help from a Vector Database."
+    human_description: str = "Answers questions about user data. Can access personal data, as it is a private store of information."
```
Comment (eob) [inline src/steamship/agents/tools/question_answering/vector_search_qa_tool.py:30]: So much better!

**Review by eob — CHANGES_REQUESTED** [review]
Change requested related to a string bug that's my fault, but rubber stamp of approval with that change made!

Thank's for catching this and also adding the unit test!!

File: src/steamship/agents/tools/question_answering/vector_search_learner_tool.py (hunk 1 above)
Comment (douglas-reid) [reply src/steamship/agents/tools/question_answering/vector_search_learner_tool.py]: Should be fixed now.

**Review by eob — APPROVED** [review]
LGTM!

**Pull request merged**